cmake/external/dnnl.cmake

```diff
@@ -15,6 +15,13 @@ else()
   endif()  
 endif()
 
+if(onnxruntime_USE_DNNL AND ((onnxruntime_DNNL_THREAD STREQUAL "eigen") OR (onnxruntime_DNNL_THREAD STREQUAL "ort")))
+  message(STATUS "dnnl is set to use a custom threadpool thread")
+  set(DNNL_THREAD_CMAKE_ARGS "-DDNNL_CPU_RUNTIME=THREADPOOL")
+else()
+  message(STATUS "dnnl is set to use openmp")
+endif()
+
```

Comment:
Does it make sense to add the user to build onednn with either OMP or TBB in case they don't use `threadpool`?

Reply:
@jywu-msft could you run /azp run Linux DNNL CI Pipeline again please?